### PR TITLE
Mes 6303 office screen 2

### PR DESCRIPTION
**Pull request by andrewsetterfield**
## Description

Migrate office page from des3 to des4 and move common logic to base page

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
Screenshot 2021-09-28 at 10 22 16
Screenshot 2021-09-28 at 10 22 31
Screenshot 2021-09-28 at 10 22 41
Screenshot 2021-09-28 at 10 23 09

### CODE REVIEW

File: src/app/pages/office/cat-b/office.cat-b.page.ts [hunk 1]
```diff
@@ -1,58 +1,276 @@
...
+    } else if (drivingFaultComment.source === CommentSource.CONTROLLED_STOP) {
+      this.store$.dispatch(AddControlledStopComment(drivingFaultComment.comment));
+
+    } else if (drivingFaultComment.source === CommentSource.VEHICLE_CHECKS) {
+      this.store$.dispatch(AddShowMeTellMeComment(drivingFaultComment.comment));
+    }
+
   }
 
+  shouldDisplayDrivingFaultComments = (data: CatBUniqueTypes.TestData): boolean => {
```
Comment (matthew2564) [inline src/app/pages/office/cat-b/office.cat-b.page.ts:270]: Can we add a TODO to create a new method inside the ```faultCountProvider```, something like getMaximumFaultSumCount. The new method could take in the maximum count i.e. 15 in this case and the category (assume all categories have similar calls so would be nice to centralise)

File: src/components/common/candidate-section/candidate-section.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <ion-row class="mes-full-width-card" id="candidate-section">
   <ion-col>
-    <h2 id="candidate-name">{{ candidateName }}</h2>
+    <h2 id="candidate-name">Mr Iva Gotta </h2>
```
Comment (matthew2564) [inline src/components/common/candidate-section/candidate-section.html]: please undo this

Comment (andrewsetterfield) [reply src/components/common/candidate-section/candidate-section.html]: How the hell did that get into des4...completely baffled.  will revert.

File: src/app/pages/office/cat-b/office.cat-b.page.ts (hunk 1 above)
Comment (andrewsetterfield) [reply src/app/pages/office/cat-b/office.cat-b.page.ts:270]: added mes-7281

File: src/components/common/candidate-section/candidate-section.html (hunk 2 above)
Comment (andrewsetterfield) [reply src/components/common/candidate-section/candidate-section.html]: reverted

File: src/app/pages/office/components/vehicle-checks/vehicle-checks-office-card.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            <ion-col size="6" class="ion-align-self-start">
+                <div class="counter-icon">
+                    <img *ngIf="!questionHasFault(check)" src="assets/imgs/greenCorrectAnswer.png" alt="Correct Answer"
+                        class="correct-icon icon-image" />
+                    <img *ngIf="questionHasFault(check)" src="assets/imgs/redWrongAnswer.png" alt="Wrong Answer"
+                        class="wrong-icon icon-image" />
+                </div>
+            </ion-col>
+            <ion-col size="3" class="ion-no-padding"></ion-col>
+            <ion-col align-self-start>
```
Comment (matthew2564) [inline src/app/pages/office/components/vehicle-checks/vehicle-checks-office-card.html]: ```class="ion-align-self-start"```

File: src/app/shared/classes/test-flow-base-pages/office/__tests__/office-base-page.spec.ts [hunk 4]
```diff
@@ -76,4 +135,60 @@ describe('OfficeBasePageComponent', () => {
...
+  describe('defer', () => {
+    it('should call popToRoot and dispatch two actions', async () => {
+      spyOn(basePageComponent, 'popToRoot');
+      await basePageComponent.defer();
+      expect(basePageComponent.popToRoot).toHaveBeenCalled();
+      expect(store$.dispatch).toHaveBeenCalledWith(SavingWriteUpForLater());
+      expect(store$.dispatch).toHaveBeenCalledWith(PersistTests());
+    });
+  });
 });
```
Comment (matthew2564) [inline src/app/shared/classes/test-flow-base-pages/office/__tests__/office-base-page.spec.ts:267]: There are a lot of untested methods in this file - might be worth leaving a TODO to add some coverage. I can see some of the tests in mes-mobile-app exist but within each spec so would be the case of just porting those over for some of these methods e.g.

```additionalInformationChanged``` has a test within ```src/pages/office/cat-a-mod2/__tests__/office.cat-a-mod2.page.spec.ts```

```candidateDescriptionChanged``` has a test within ```src/pages/office/cat-a-mod2/__tests__/office.cat-a-mod2.page.spec.ts```

```routeNumberChanged``` has a test within ```src/pages/office/cat-a-mod2/__tests__/office.cat-a-mod2.page.spec.ts```

```independentDrivingChanged``` has a test within ```src/pages/office/cat-a-mod2/__tests__/office.cat-a-mod2.page.spec.ts```

```identificationChanged``` has a test within ```src/pages/office/cat-a-mod2/__tests__/office.cat-a-mod2.page.spec.ts```

Comment (andrewsetterfield) [reply src/app/shared/classes/test-flow-base-pages/office/__tests__/office-base-page.spec.ts:267]: I didnt like leaving the file untested, but following our agreed pattern of filling in the tech debt at a later date.  Will add a TODO to that effect and story to match.

File: src/app/pages/office/components/vehicle-checks/vehicle-checks-office-card.html (hunk 3 above)
Comment (andrewsetterfield) [reply src/app/pages/office/components/vehicle-checks/vehicle-checks-office-card.html]: have updated. cant add screen shot as i dont believe this is part of cat b.

File: src/app/shared/classes/test-flow-base-pages/office/__tests__/office-base-page.spec.ts (hunk 4 above)
Comment (matthew2564) [reply src/app/shared/classes/test-flow-base-pages/office/__tests__/office-base-page.spec.ts:267]: Its not tech debt if the tests already exist surely? Agree with TODO re the other methods as wouldn't want you to waste time making new tests

**Review by matthew2564 — APPROVED** [review]

**Review by gittins7 — APPROVED** [review]

**Pull request merged**